java-frontend/src/main/java/org/sonar/java/reporting/InternalJavaIssueBuilder.java

```diff
@@ -227,32 +232,40 @@ public void report() {
       }
     }
 
-
-    if (!quickFixes.isEmpty() && sonarComponents.isQuickFixCompatible()) {
-      addQuickFixes(ruleKey.get(), (NewSonarLintIssue) newIssue);
+    final List<JavaQuickFix> flatQuickFixes = quickFixes.stream()
+      .flatMap(s -> s.get().stream())
+      .collect(Collectors.toList());
```

Comment:
As discussed, we are always computing the quick fixes here, even when we have nothing to do with them.
I agree that we will eventually always compute them for all products and delaying the computation is just a temporary minor performance gain. Still, we already have the logic in place, we can benefit from it by refactoring the logic here.
Even if it is less readable, I think it is worth it.